### PR TITLE
Index, Show, New CHAIRS

### PR DESCRIPTION
**Pull request by dpoint01**
Took care of basic set up of index, show, new for chairs

---

**Remains to be done:**

**View all Chairs**
As a user I want to be able to see all the previous posts so that I can know what chairs are out there

**Acceptance Criteria**
- Check if signed in
- view the average rating of each chair (All ratings on that page are the average of all the ratings from the reviews)
- All posts should be ranked according to their rating
- Each picture/or name should take me to the show page of that chair so that I can see its details
- I should be able to search for a specific chair.
- I should be able to navigate through the different pages.

**Add a Chair**
As a user I want to be able to add a chair to the website 
so that I can see it in the index page

**Acceptance Criteria**
- I should be signed in!
- I should be alerted if any field is filled-in in the wrong fashion
- I should be notified by email everytime a user reviews my post.

---

**Achieved:**

**View all Chairs**

As a user
I want to be able to see all the previous posts 
so that I can know what chairs are out there

**Acceptance Criteria**
- I want to be able to see the picture (if there is one), the name, and the rating of each chair
- (have a link for this instead of picture) Each picture/or name should take me to the show page of that chair so that I can see its details.

**Add a Chair**

As a user 
I want to be able to add a chair to the website 
so that I can see it in the index page

**Acceptance Criteria**
- I can add a chair with: a new picture, a name, a breif desciption(optional), and its characteristics (some will be tick    form, other fill-ins)
- If the chair is successfully saved, I should be alerted so and redirected to the newly chair show page.

Note: Still need to replace categories with tags so that user can select several categories for one chair instead of one currently

### CODE REVIEW

File: README.rdoc [hunk 1]
```diff
@@ -0,0 +1,28 @@
+== README
```
Comment (radavis) [inline README.rdoc]: This file can be deleted.

File: app/controllers/chairs_controller.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
+class ChairsController < ApplicationController
+  def index
+    @chairs = Chair.all.limit(25)
```
Comment (radavis) [inline app/controllers/chairs_controller.rb]: You can change this to `@chairs = Chair.limit(25)`. Queries to the .all method will become expensive when the database grows.

File: app/controllers/chairs_controller.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    @review = Review.new
+  end
+
+  def new
+    @chair = Chair.new
+  end
+
+  def create
+    @chair = Chair.new(chair_params)
+    #binding.pry
```
Comment (radavis) [inline app/controllers/chairs_controller.rb]: remove commented code

File: app/views/chairs/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,9 @@
+<h1> <em> Fauteuil's </em> Selection of Chairs </h1>
+
+<ul>
+  <% @chairs.each do |chair| %>
+    <li> <%= chair.name %> || <%= chair.location %> || <%= chair.picture %> || <%= link_to "==>", chair_path(chair.id)%> </li>
```
Comment (radavis) [inline app/views/chairs/index.html.erb]: Break up long lines of code over multiple lines.

File: app/views/chairs/new.html.erb [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  <%= f.text_field :name , :placeholder => "e.g. stool" %><br>
+
+  <%= f.label :manufacturer %>:
+  <%= f.text_field :manufacturer,:placeholder => "e.g. Ikea" %><br>
+
+  <%= f.label :location %>:
+  <%= f.text_field :location,:placeholder => "where you found this chair" %><br>
+
+  <%= f.label :description %>:
+  <%= f.text_area :description, :placeholder => "e.g. Amazing back support, blends in with my red curtains, etc..." %><br>
```
Comment (radavis) [inline app/views/chairs/new.html.erb]: Use the updated hash syntax.

```
<%= f.text_area :description, placeholder: "e.g. Amazing back support, blends in with my red curtains, etc..." %>
```

File: app/views/chairs/show.html.erb [hunk 6]
```diff
@@ -0,0 +1,11 @@
+<h2> <em> <%=@chair.name %> </em> </h2>
```
Comment (radavis) [inline app/views/chairs/show.html.erb]: Use correct spacing. Nest tags.

```
<h2>
  <em><%= @chair.name %></em>
</h2>

```

File: db/seeds.rb [hunk 7]
```diff
@@ -5,3 +5,39 @@
 #
 #   cities = City.create([{ name: 'Chicago' }, { name: 'Copenhagen' }])
 #   Mayor.create(name: 'Emanuel', city: cities.first)
+
+categories = ['Stool',
```
Comment (radavis) [inline db/seeds.rb]: format your array this way: 

```
categories = [
  'Stool',
  'Barber Chair',
  'Broken Chair',
  'Desk Chair',
  ...
]
```

File: spec/models/chair_spec.rb [hunk 8]
```diff
@@ -1,5 +1,23 @@
...
 
-RSpec.describe Chair, :type => :model do
-  pending "add some examples to (or delete) #{__FILE__}"
+# RSpec.describe Chair, :type => :model do
+describe Chair do
+
+  it { should validate_uniqueness_of(:name) }
+
+
+  # it 'validates uniqueness of chair name' do
```
Comment (radavis) [inline spec/models/chair_spec.rb]: remove commented code.
add tests for all fields that have null constraints:

```
it { should have_valid(:name).when('Aeron', 'Stool') }
it { should_not have_valid(:name).when(nil, '') }
```

You may have to add the `gem 'valid_attribute'` to the gemfile and bundle.

File: db/seeds.rb [hunk 9]
```diff
@@ -5,3 +5,39 @@
...
+              'Antique Chair',
+              'Modern Chair',
+              'Captains chair',
+              'Kitchen Chair',
+              'Special Chair',
+              'Living Room Chair',
+              'Mobile Chair'
+              ]
+
+categories.each do |cat|
```
Comment (radavis) [inline db/seeds.rb]: category is a better variable name than cat
